lookout/style/format/annotations/annotations.py

```diff
@@ -0,0 +1,196 @@
+
+        :param start: Annotation start.
+        :param stop: Annotation end.
+        :param uast: UAST.
+        """
+        super().__init__(start, stop, uast)
+        self._parents = get_uast_parents(uast)
+
+    uast = property(lambda self: self._node)
+
+    parents = property(lambda self: self._parents)
+
+
+class TokenAnnotation(Annotation):
+    """Annotation for virtual token in the сode."""
+
+    def __init__(self, start: int, stop: int,
+                 uast_annotation: Optional[UASTNodeAnnotation] = None):
+        """
+        Initialization.
+
+        :param start: Annotation start.
+        :param stop: Annotation end.
+        :param uast_annotation: Related UASTNodeAnnotation Annotation if applicable.
+        """
+        super().__init__(start, stop)
+        self._uast_annotation = uast_annotation
+
+    uast_annotation = property(lambda self: self._uast_annotation)
+
+    @property
+    def node(self) -> "bblfsh.Node":
+        """
+        Get UAST Node from related UASTNodeAnnotation.
+
+        :return: related bblfsh UAST Node. None if there is no related annotation.
+        """
+        return self._uast_annotation.node if self._uast_annotation else None
+
+    @property
+    def has_node(self) -> bool:
+        """Check if token annotation has related UAST node annotation."""
+        return self._uast_annotation is not None
+
+
+class AtomicTokenAnnotation(TokenAnnotation):
+    """Annotation for undividable tokens generated by FeatureExtractor._classify_vnodes()."""
+
+    def to_token_annotation(self) -> TokenAnnotation:
+        """Convert to token annotation."""
+        return TokenAnnotation(self.start, self.stop, self._uast_annotation)
+
+
+class RawTokenAnnotation(TokenAnnotation):
+    """Annotation for raw tokens generated by FeatureExtractor._parse_file()."""
+
+    def to_atomic_token_annotation(self) -> AtomicTokenAnnotation:
+        """Convert to atomic token annotation."""
+        return AtomicTokenAnnotation(self.start, self.stop, self._uast_annotation)
+
+
+class LanguageAnnotation(Annotation):
+    """Language of the file annotation."""
+
+    def __init__(self, start: int, stop: int, language: str):
+        """Init."""
+        super().__init__(start, stop)
+        self._language = language
+
+    language = property(lambda self: self._language)
+
+
+class PathAnnotation(Annotation):
+    """File language annotation."""
+
+    def __init__(self, start: int, stop: int, path: str):
+        """Init."""
+        super().__init__(start, stop)
+        self._path = path
+
+    path = property(lambda self: self._path)
+
+
+class TargetAnnotation(Annotation):
+    """Target for model prediction annotation."""
+
+    def __init__(self, start: int, stop: int, target: Tuple[int, ...]):
+        """Init."""
+        super().__init__(start, stop)
+        self._target = target
+
+    target = property(lambda self: self._target)
+
+
+class AtomicTargetAnnotation(TargetAnnotation):
```

Comment:
Let's use "class" instead of "atomic target" and "label" instead of "target" everywhere. This corresponds to the terms in our paper.